packages/itwinui-react/src/core/DatePicker/DatePicker.tsx

```diff
@@ -595,20 +643,32 @@ export const DatePicker = React.forwardRef((props, forwardedRef) => {
           onKeyDown={handleCalendarKeyDown}
           role='listbox'
           aria-labelledby={dateTableId}
+          {...calendarProps}
+          className={calendarProps?.className}
         >
           {weeks.map((weekDays, weekIndex) => {
             return (
               <Box
+                as='div'
+                {...calendarWeekProps}
                 key={`week-${displayedMonthIndex}-${weekIndex}`}
-                className='iui-calendar-week'
+                className={cx(
+                  'iui-calendar-week',
+                  calendarWeekProps?.className,
+                )}
               >
                 {weekDays.map((weekDay, dayIndex) => {
                   const dateValue = weekDay.getDate();
                   const isDisabled = isDateDisabled?.(weekDay);
                   return (
                     <Box
+                      as='div'
+                      {...calendarDayProps}
```

Comment:
the `calendarDayProps` and `className` props need to be moved down a few lines, similar to how i described [earlier](https://github.com/iTwin/iTwinUI/pull/1466#discussion_r1304380707).